src/Utilities/ArchiveTools.ts

```diff
@@ -0,0 +1,43 @@
+import { debug } from "@actions/core";
+import { exec } from "@actions/exec";
+import { which } from "@actions/io";
+import PowerShellToolRunner from "./PowerShellToolRunner";
+
+export class ArchiveTools {
+    private use7Zip: boolean;
+
+    constructor(use7Zip = false) {
+        this.use7Zip = use7Zip;
+    }
+
+    public async unzip(zipPath: string, destination: string) {
+        if (this.use7Zip) {
+            await this.unzipUsing7Zip(zipPath, destination);
+        } else {
+            await this.unzipUsingPowerShell(zipPath, destination)
```

Comment:
are we using powershell for unzipping on linux?

Reply:
Yes
pwsh Expand-Archive in case of linux
7zip in case of windows

Reply:
why can't default unzip module on linux be used?

Reply:
Preferred pwsh since it is platform neutral and seems more future proof in case we plan to support more platform. Would have used the same for windows but avoid due to huge perf diff.
In linux/mac there is not perf impact in using pwsh over unzip.
Please let me know in case you feel otherwise.